GiniVision/Classes/ReviewContainerViewController.swift

```diff
@@ -78,10 +77,35 @@ internal class ReviewContainerViewController: UIViewController, ContainerViewCon
         displayContent(contentController)
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Configure back button. Needs to be done here because otherwise the navigation controller would be nil
+        guard let _ = navigationItem.leftBarButtonItem else {
+            setupLeftNavigationItem(usingResources: backButtonPreferredResource(), selector:#selector(back))
+            return
+        }
+    }
+    
     @IBAction func back() {
         let delegate = (navigationController as? GiniNavigationViewController)?.giniDelegate
         delegate?.didCancelReview?()
-        _ = navigationController?.popViewController(animated: true)
+        if let navVC = navigationController {
+            if navVC.viewControllers.count > 1 {
```

Comment:
This check for > 1 view controllers in the stack is used in two places and should be extracted into a method to DRY it up and also make the code more readable.

Reply:
I thought about it but I wasn't sure that it was worth for only one line. But since one of the methods has an `else`,  I cannot extract it into a `isFirstViewControllerInTheStack` method/computedProperty.

So, do you think that It is worth to move `navVC.viewControllers.count > 1` to a method?

Reply:
What about this? I added some guards to make it more obvious that a navigation controller is required.
```
override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        setupLeftBarButton()
    }
    
    fileprivate func setupLeftBarButton() {
        guard let navController = navigationController else {
            return
        }
        
        // Configure back button. Needs to be done here because otherwise the navigation controller would be nil
        if navigationItem.leftBarButtonItem == nil {
            setupLeftNavigationItem(usingResources: backButtonPreferredResource(forNavController: navController), selector:#selector(back))
        }
    }
    
    @IBAction func back() {
        let delegate = (navigationController as? GiniNavigationViewController)?.giniDelegate
        delegate?.didCancelReview?()
        
        handleBackPressed()
    }
    
    fileprivate func handleBackPressed() {
        guard let navController = navigationController else {
            return
        }
    
        if firstViewController(inNavController: navController) {
            navController.dismiss(animated: true, completion: nil)
        } else {
            _ = navController.popViewController(animated: true)
        }
    }
    
    fileprivate func backButtonPreferredResource(forNavController navController:UINavigationController) -> PreferredButtonResource {
        if firstViewController(inNavController: navController) {
            return closeButtonResources
        }
        return backButtonResources
    }
    
    fileprivate func firstViewController(inNavController navController:UINavigationController ) -> Bool {
        if navController.viewControllers.count == 1 {
            return true
        }
        return false
    }
```